### DISCUSSION TITLE
v4.43.0

### DISCUSSION THREAD

**Discussion by xiaokangwang** [v2fly/v2ray-core#1315] [General]
## Features
* **Shadowsocks** Send Shadowsocks handshake with payload data if available (#1285 Thanks @nekohasekai)
* **TFO**  Custom TCP Fast Open queue length support (#1293 Thanks @AkinoKaede)

## Fixes
* Trojan fallback cannot get ALPN (#1286 Thanks @AkinoKaede)
* QueryStrategy ignored (#1285 Thanks @nekohasekai)
* UDP connection transport connection terminated unnecessarily (#1305 Thanks @ovear)
## Chores
* refactor: move from io/ioutil to io and os package (#1298 Thanks @Juneezee)
* Fix some tests to use udp.PickPort() (#1302 Thanks @yuhan6665 )
* Fix flaky TestVMessDynamicPort (#723) (#1289 Thanks @yuhan6665)

## !!! Important Message !!!

V2Ray(V2Fly) will pre-release its next major version: V2Ray V5. In addition to functionality improvements, it will include a new configuration format and infrastructure changes that streamline the development of new protocols and functionalities.

The V4 version of the V2Ray will then enter maintenance mode. No additional features will be added by core developers, while contributors may still send pull requests for new features. It will receive bug fixes and security updates from core developers for a limited period until the full release of the V5 version. Contributors are encouraged to fork and base their changes from V5 version branch, instead of V4 version to reduce merge conflict. If you have already started the development of a change based on V4 version, you may send your pull request to V4 branch for a limited period. The core developers will cherry-pick that change into V5 version on your behave.

The pre-release version of V2Ray V5 may still contain bugs or inconsistencies. Some breaking updates to it are expected. You will need to change your configuration or codebase alongside us if you switch to V2Ray V5 now.

When V2Ray V5 pre-releases the master branch will switch to V5 version of the codebase, and the new V5 binary will be pre-released in the Github Release. They are not suitable for an automatic update from the V4 version.

## Notice

Due to increase in size of the `geoip.dat` file recently, devices with insufficient ROM/RAM are experiencing difficulties in using V2Ray. The solution is as follows:

* For **RAM** insufficient devices: Enable the Geodata loader optimized for memory-constrained devices by setting the environment variable `V2RAY_CONF_GEOLOADER` to value `memconservative`. For more details, see [documentation](https://www.v2fly.org/config/env.html#geodata-%E6%96%87%E4%BB%B6%E5%8A%A0%E8%BD%BD%E5%99%A8).
* For **ROM** insufficient devices:
  * Use the newly added GeoIP file `geoip-only-cn-private.dat` in the zip package or download it from [release page](https://github.com/v2fly/geoip/releases), which only contains GeoIP list `geoip:cn` and `geoip:private`, or
  * Customize your own GeoIP file via project [v2fly/geoip](https://github.com/v2fly/geoip).

<hr /><em>This discussion was created from the release <a href='https://github.com/v2fly/v2ray-core/releases/tag/v4.43.0'>v4.43.0</a>.</em>

**Comment by ganli** [comment]
Hi, 

I've downloaded v2ray (os linux64，arm5，arm8) from 4.43 release page, but the command "v2ray --version" shows the version is 4.42.2.
Please check if this is packaging issue or the version number is not updated when release.

BRs,

**Reply by AkinoKaede** [reply]
No, the `version` was changed. e621175927767972831c943e34c216c86b59f086

**Reply by ganli** [reply]
Thanks, I made one stupid mistake. Should use "./v2ray --version".
